docs/applications/containers/linode_kubernetes_guide.md

```diff
@@ -0,0 +1,389 @@
+
+        apt install ebtables ethtool
+
+**Centos:**
+
+        yum install ebtables ethtool
+
+### Install Docker:
+
+**Ubuntu:**
+
+    apt-get update && apt-get install -y curl apt-transport-https
+    curl -fssl https://download.docker.com/linux/ubuntu/gpg | apt-key add -
+    cat <<eof >/etc/apt/sources.list.d/docker.list
+    deb https://download.docker.com/linux/$(lsb_release -si | tr '[:upper:]' '[:lower:]') $(lsb_release -cs) stable
+    eof
+    apt-get update && apt-get install -y docker-ce=$(apt-cache madison docker-ce | grep 17.03 | head -1 | awk '{print $3}')
+
+**Centos:**
+
+    yum install -y docker
+    systemctl enable docker && systemctl start docker
+
+
+Check your Docker configuration works correctly:
+
+    docker run hello-world
+
+
+### Install kubeadm kubectl and kubelet
+
+**Ubuntu:**
+
+{:.output}
+~~~
+curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -
+echo 'deb http://apt.kubernetes.io/ kubernetes-xenial main' | sudo tee /etc/apt/sources.list.d/kubernetes.list
+sudo apt update
+sudo apt install -y kubelet kubeadm kubectl
+~~~
+
+**Centos:**
+
+{:.output}
+~~~
+cat <<eof > /etc/yum.repos.d/kubernetes.repo
+[kubernetes]
+name=kubernetes
+baseurl=https://packages.cloud.google.com/yum/repos/kubernetes-el7-x86_64
+enabled=1
+gpgcheck=1
+repo_gpgcheck=1
+gpgkey=https://packages.cloud.google.com/yum/doc/yum-key.gpg
+        https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
+eof
+setenforce 0
+yum install -y kubelet kubeadm kubectl
+systemctl enable kubelet && systemctl start kubelet
+~~~
+
+## Kubernetes Master and Slave
+
+### Set up the master node
+
+1.  On the master node initialize your cluster using its private IP:
+
+        kubeadm init  --pod-network-cidr=192.168.0.0/16 --apiserver-advertise=<private IP>
+
+    If you encounter a warning stating that swap is enabled, return to the section regarding swap memory.
+
+    {:.output}
+    ~~~
+    To start using your cluster, you need to run (as a regular user):
+
+      mkdir -p $HOME/.kube
+      sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
+      sudo chown $(id -u):$(id -g) $HOME/.kube/config
+
+    You should now deploy a pod network to the cluster.
+    Run "kubectl apply -f [podnetwork].yaml" with one of the options listed at:
+      http://kubernetes.io/docs/admin/addons/
+
+    You can now join any number of machines by running the following on each node
+    as root:
+
+      kubeadm join --token 921e92.d4582205da623812 <private IP>:6443 --discovery-token-ca-cert-hash sha256:bd85666b6a97072709b210ddf677245b4d79dab88d61b4a521fc00b0fbcc710c
+    ~~~
+
+    You can create a new token your master host. Enter `kubeadm token create` to generate a new token and pass it as the token parameter in the above command. This token will also only last for 24 hours unless you pass it with the `--ttl infinity` flag
+
+2.  Theres a second set of commands above that which describe how to configure the `kubectl` tool.
```

Comment:
@GuessWhoSamFoo should this be run from the master or a node?

Reply:
Token generation happens on a master node

Reply:
thanks, @GuessWhoSamFoo 

I'll edit that in and push over the copy edits